builtin/logical/database/backend.go

```diff
@@ -76,13 +76,60 @@ func Backend(conf *logical.BackendConfig) *databaseBackend {
 		Secrets: []*framework.Secret{
 			secretCreds(&b),
 		},
-		Clean:       b.closeAllDBs,
-		Invalidate:  b.invalidate,
-		BackendType: logical.TypeLogical,
+		Clean:             b.closeAllDBs,
+		Invalidate:        b.invalidate,
+		BackendType:       logical.TypeLogical,
+		WALRollbackMinAge: 30 * time.Second,
 	}
 
 	b.logger = conf.Logger
 	b.connections = make(map[string]*dbPluginInstance)
+
+	// Always populate the queue to guard against nil pointers.
+	b.credRotationQueue = queue.NewTimeQueue()
+
+	// But only execute cred rotation if we're the leader.
+	// TODO what if leadership changes during the lifecycle of the application?
+	replicationState := conf.System.ReplicationState()
+	if replicationState.IsLeader() {
```

Comment:
We also need to include a queue if this is a local mount, does this check include that?

Reply:
I don't think local mounts are handled yet (@tyrannosaurus-becks can elaborate).

Reply:
Yeah it checks them currently! However I haven't tested the logic IRL, it does need that. The logic checks if replication is disabled, which is how it's attempting to derive whether it's just a standalone. Not certain if that's what you mean by "local" mounts - if it's synonymous with "standalone" or if you also mean in a cluster but is the leader. The logic within the method definitely needs checking.